cpp/src/parquet/encoding.cc

```diff
@@ -2100,9 +2100,12 @@ class DictByteArrayDecoderImpl : public DictDecoderImpl<ByteArrayType>,
 
 template <typename DType>
 class DeltaBitPackEncoder : public EncoderImpl, virtual public TypedEncoder<DType> {
+  static_assert(std::is_same_v<int32_t, typename DType::c_type> ||
+                std::is_same_v<int64_t, typename DType::c_type>);
   // Maximum possible header size
   static constexpr uint32_t kMaxPageHeaderWriterSize = 32;
```

Comment:
It would be better to make them configurable. Either at writer level (via `WriterProperties`) or column level (via `ColumnProperties`). Otherwise user does not have a way to tune performance or mitigate regression at the run time.

Reply:
Yes, something like `column_properties->set_delta_bit_pack_params(a, b)` ?

Reply:
Would we want to expose this in other language bindings too? (e.g. Python, R)

Reply:
It's a bit hacking here. Let me have a try...

Reply:
Perhaps that would be better as a separate issue.